narwhal/network/src/lib.rs

```diff
@@ -56,13 +56,9 @@ pub fn multiaddr_to_address(
     let mut iter = multiaddr.iter();
 
     match (iter.next(), iter.next()) {
-        (Some(Protocol::Ip4(ipaddr)), Some(Protocol::Tcp(port) | Protocol::Udp(port))) => {
-            Ok((ipaddr, port).into())
-        }
-        (Some(Protocol::Ip6(ipaddr)), Some(Protocol::Tcp(port) | Protocol::Udp(port))) => {
-            Ok((ipaddr, port).into())
-        }
-        (Some(Protocol::Dns(hostname)), Some(Protocol::Tcp(port) | Protocol::Udp(port))) => {
+        (Some(Protocol::Ip4(ipaddr)), Some(Protocol::Udp(port))) => Ok((ipaddr, port).into()),
+        (Some(Protocol::Ip6(ipaddr)), Some(Protocol::Udp(port))) => Ok((ipaddr, port).into()),
+        (Some(Protocol::Dns(hostname)), Some(Protocol::Udp(port))) => {
```

Comment:
Since this removes the ability to interpret a TCP multiaddr from this utility, I feel we should at least remark this in a doc comment, stg like "this expects a UDP endpoint in multiaddr format".